pkg/policy/rule.go

```diff
@@ -512,6 +535,23 @@ func (r *rule) resolveCIDRPolicy(ctx *SearchContext, state *traceState, result *
 		}
 	}
 
+	for _, ingressRule := range r.IngressDeny {
+		// TODO (ianvernon): GH-1658
+		var allCIDRs []api.CIDR
```

Comment:
I think we can preallocate the minimum size of this slice here:

```suggestion
		allCIDRs := make([]api.CIDR, 0, len(ingressRule.FromCIDR))
```